File: fibonacci/fibonacci.cpp
/*
 * Fibonacci implementation for Rosetta Stone project
 */

#include <iostream>
using namespace std;

void fibonacci(void)
{
    int prev = 0;
    int cur = 1;
    int n_cur;

    for (int i = 0; i < 10; i++) {
        n_cur = cur + prev;
        prev = cur;
        cur = n_cur;

        cout<<cur<<" ";
    }
}

int main(void)
{
    fibonacci();
}

File: fibonacci/fibonacci.kt
fun main(args: Array<String>) {
    fibonacci();
}

fun fibonacci(): Unit {
    var prev = 0
    var actual = 1
    var aux: Int

    for (q in 0 until 10) {
        aux = actual
        actual = prev + aux
        prev = aux
        println(actual)
    }
}

